apollo-ios-codegen/Sources/ApolloCodegenLib/CodegenConfiguration/ApolloCodegenConfiguration.swift

```diff
@@ -321,12 +326,87 @@ public struct ApolloCodegenConfiguration: Codable, Equatable {
           self = .embeddedInTarget(name: name, accessModifier: accessModifier)
 
         case .swiftPackageManager:
-          self = .swiftPackageManager
+          self = .swiftPackage(apolloSDKVersion: .default)
+          
+        case .swiftPackage:
+          let nestedContainer = try container.nestedContainer(
+            keyedBy: SwiftPackageCodingKeys.self,
+            forKey: .swiftPackage
+          )
+          
+          let apolloSDKVersion = try nestedContainer.decodeIfPresent(ApolloSDKVersion.self, forKey: .apolloSDKVersion) ?? .default
+          self = .swiftPackage(apolloSDKVersion: apolloSDKVersion)
 
         case .other:
           self = .other
         }
       }
+      
+      public enum ApolloSDKVersion: Codable, Equatable {
+        case `default`
+        case branch(name: String)
+        case commit(hash: String)
+        case custom(url: String, dependencyType: DependencyType, value: String)
```

Comment:
Also considered calling this case `fork` with same associated values

Reply:
This can only come from a forked repo right? The `url` parameter is pretty clear on that deviating from `github.com/apollographql/apollo-ios`. In that case yes, I think `fork` is the better case name.

Reply:
Yea this is really meant for use specifically with forked repos, so yea I guess the `fork` name makes more sense.